src/gensxg.c

```diff
@@ -257,6 +268,9 @@ static bool validate(const Options* opt) {
     fputs("error: -miRecordSize must be greater than 0.\n", stderr);
     valid = false;
   }
+  if (opt->integrity_mode) {
```

Comment:
This is a pretty long function, and it's hard to recognize the logic changes in the `integrity_mode` case. I'd refactor to:

```c
static bool validate_common_options(const Options* opt) {
  ...
}

static bool validate_generator_options(const Options *opt) {
  ...
}

static bool validate_options(const Options* opt) {
  bool valid = validate_common_options(opt);
  if (!opt->integrity_mode) {
    valid = valid & validate_generator_options(opt);  /* & should not be &&. */
  }
  return valid;
}
```

Note: I'm not sure about function naming.

Reply:
OK, introduced as-is.